test/xbgpu/test_engine.py

```diff
@@ -544,3 +555,78 @@ def sensor_observer(sync_sensor: aiokatcp.Sensor, sensor_reading: aiokatcp.Readi
         np.testing.assert_equal(actual_sensor_updates, expected_sensor_updates)
 
         await xbengine.stop()
+
+    async def test_saturation(
```

Comment:
While it's probably fine for now, there seems to me a lot of WET between this test and the end_to_end one. I'd feel happier if common stuff became fixtures, though I'm not sure based on only a cursory look how hard that would be.

Maybe a JIRA ticket for later.

Reply:
I thought there was a Jira ticket to factor out fixtures from xbgpu tests, but it seems to have been closed already. I'll open another one when I merge. There is definitely some stuff in these tests I want to refactor but I didn't want to disappear down the rabbit hole.

Reply:
NGC-328?
Yeah I closed that one because I'd done what I thought was expedient. It could well be that there's now more to do since stuff has been added.

Reply:
NGC-832